### DISCUSSION TITLE
Typescript broke recently in this repo

### DISCUSSION THREAD

**Discussion by unphased** [nvim-treesitter/nvim-treesitter#5247] [Q&A]
I was able to narrow down the introduction of treesitter's inability to parse my typescript code upon updating this plugin via Lazy from `a9ce9fb7` to `93ee00cd`. 

With the new commit, looking at the TSPlayground my code was previously parsing fine now has huge cascades of ERROR nodes. And the indentation does not work, and Comment.nvim is toggling HTML style comments in the typescript code, which is hilarious.

I think maybe some javascript related grammar changes may be implicated. Not sure. Has anyone observed something similar?

**Comment by unphased** [comment]
I checked out the latest commit 1a0cf38f and it won't run with `error: module 'nvim-treesitter.configs' not found` 

I guess there's some deep refactoring happening currently

**Comment by clason** [comment]
You're pulling from the main branch, which is a WIP rewrite and not meant for public use.

**Reply by unphased** [reply]
Ok but the commit I mentioned was auto grabbed by Lazy. It's causing issues.

**Reply by clason** [reply]
Then you need to complain to Lazy. This branch is not the default branch and not ready to be used.

**Reply by unphased** [reply]
Let me clarify. The issue is present on `93ee00cd`, which is currently `origin/master` here.

**Reply by unphased** [reply]
I went ahead and bisected the issue down to the commit `78b54eb7 - chore(injections)!: update injection syntax to 0.9 2023-08-12 17:34:15 +0200 (3 days ago) <Pham Huy Hoang>`.

I will try to see if reverting this commit is still possible on master. 

(Edit: Also, I tried to use the second commit before this (tag v0.9.1), but then i saw it completely broke the ability to edit cpp files. so... back to a9ce9fb7...)

**Reply by unphased** [reply]
> I will try to see if reverting this commit is still possible on master.

It is not possible to do this. Large amount of conflicts.

**Reply by unphased** [reply]
i would include my typescript code that breaks but it's slightly sensitive. I can prepare something for you if necessary. A lot of breakage with angle brackets (e.g. with types, conditions in for loops, etc)

NVIM v0.10.0-dev-2050+g86ce3878d-Homebrew
Build type: Release
LuaJIT 2.1.0-beta3

Reproduces on both macOS and Linux, my neovims are built from source, with brew on the mac and with an automated docker setup i use on linux (basically following the manual instructions)

**Reply by clason** [reply]
> Let me clarify. The issue is present on 93ee00cd, which is currently origin/master here.

Yes, I know; that is still an **intended** change that plugins will have to adapt to. I don't know how much clearer we can make that nvim-treesitter (just as treesitter support in general) is experimental and _will_ have breaking changes. If you (as user or plugin dev) can't deal with that, you shouldn't depend on it (yet).
